libtiledbsoma/src/utils/arrow_adapter.h

```diff
@@ -229,6 +229,22 @@ class ArrowAdapter {
     static std::unique_ptr<ArrowSchema> arrow_schema_from_tiledb_array(
         std::shared_ptr<Context> ctx, std::shared_ptr<Array> tiledb_array);
 
+    /**
+     * @brief Create an ArrowSchema from TileDB Dimension
+     *
+     * @return ArrowSchema
+     */
+    static std::unique_ptr<ArrowSchema> arrow_schema_from_tiledb_dimension(
```

Comment:
I wish you had made these helpers as a separate PR. As an ongoing ask I am continuing to ask for split-out PRs.

This PR is only 300 lines, which isn't a lot, but, all the codecov warnings are making this hard to read -- as if it _had_ been a high-line-count PR.

Reply:
I ask you to split this not due to high line count but because you've got two different things going on. Pulling out helper functions is distinct from the goal of the PR (hiding internal dimensions) and is worth a separate PR. Then you can stack the dimension-hiding on top of it as a second PR.

Reply:
Specifically, please split out a PR with only the `arrow_adapter.h` and `arrow_adapter.cc` mods. That, I will understand quickly, and will be able to review quickly.